File: app/src/main/java/org/cuieney/videolife/presenter/contract/EssayHomeContract.java
package org.cuieney.videolife.presenter.contract;

import org.cuieney.videolife.common.base.BasePresenter;
import org.cuieney.videolife.common.base.BaseView;
import org.cuieney.videolife.entity.MusicListBean;
import org.cuieney.videolife.entity.YiRenBean.DataBean;

import java.util.List;

/**
 * Created by cuieney on 2017/5/17.
 */
public interface EssayHomeContract {
    interface View extends BaseView {
        void showContent(List<DataBean> dataBeanList);
        void error(Throwable throwable);
    }

    interface Presenter extends BasePresenter<EssayHomeContract.View> {
        void getEssayData(int id);

    }
}

File: app/src/main/java/org/cuieney/videolife/kotlin/presenter/contract/EssayHomeContract.kt
package org.cuieney.videolife.kotlin.presenter.contract

import org.cuieney.videolife.entity.YiRenBean.DataBean
import org.cuieney.videolife.kotlin.base.BasePresenter
import org.cuieney.videolife.kotlin.base.BaseView


/**
 * Created by cuieney on 2017/5/17.
 */
interface EssayHomeContract {
    interface View : BaseView {
        fun showContent(dataBeanList: List<DataBean>)
        fun error(throwable: Throwable)
    }

    interface Presenter : BasePresenter {
        fun getEssayData(id: Int,mView: View)
    }
}
